static/app/components/calendar/dateRangePicker.tsx

```diff
@@ -0,0 +1,45 @@
+import {useCallback, useMemo} from 'react';
+import {DateRange, DateRangeProps, Range, RangeKeyDict} from 'react-date-range';
+
+import {callIfFunction} from 'sentry/utils/callIfFunction';
+
+import CalendarStylesWrapper from './calendarStylesWrapper';
+
+export type DateRangePickerProps = Omit<DateRangeProps, 'ranges' | 'onChange'> & {
+  onChange: (range: Range) => void;
+  endDate?: Date;
+  startDate?: Date;
+};
+
+type RangeSelection = {primary: Range};
+
+const PRIMARY_RANGE_KEY = 'primary';
+
+function isRangeSelection(rangesByKey: RangeKeyDict): rangesByKey is RangeSelection {
+  return rangesByKey?.[PRIMARY_RANGE_KEY] !== undefined;
+}
+
+const DateRangePicker = (props: DateRangePickerProps) => {
+  const onChange = useCallback(
+    (rangesByKey: RangeKeyDict) => {
+      if (!isRangeSelection(rangesByKey)) {
+        return;
+      }
+
+      callIfFunction(props.onChange, rangesByKey[PRIMARY_RANGE_KEY]);
```

Comment:
You can just
```suggestion
      props.onChange?.(rangesByKey[PRIMARY_RANGE_KEY]);
```

Reply:
Saw this being used in other components but if we don't care for the type safety anymore (because typescript) I like the other way better too

Reply:
Yeah typescript should help us here